docs/user-guide/multiple-ingress.md

```diff
@@ -1,7 +1,63 @@
 # Multiple Ingress controllers
 
-If you're running multiple ingress controllers, or running on a cloud provider that natively handles ingress such as GKE,
-you need to specify the annotation `kubernetes.io/ingress.class: "nginx"` in all ingresses that you would like the ingress-nginx controller to claim.
+By default, deploying multiple Ingress controllers (e.g., `ingress-nginx` & `gce`) will result in all controllers simultaneously racing to update Ingress status fields in confusing ways.
+
+To fix this problem, you can set either the `ingressClassName` (preferred) or the `kubernetes.io/ingress.class` annotation (in deprecation) on your Ingress resources.
+
+## Setting ingressClassName
+
+If all ingress controllers respect IngressClasses (e.g. multiple instances of ingress-nginx v1.0), ensure that the `--controller-class` flag, the IngressClass name, and the `ingressClassName` field are in sync:
+
+```yaml
+# ingress-nginx Deployment/Statfulset
+spec:
+  template:
+     spec:
+       containers:
+         - name: nginx-ingress-internal-controller
+           args:
+             - /nginx-ingress-controller
+             - '--controller-class=internal-nginx'
```

Comment:
This is wrong :)

This should reflect .spec.controller value in IngressClass object

Reply:
Ah good catch. I reworked the section, and explicitly made the end part of --controller-class flag not match the IngressClass name to help people not make my same mistake :)